jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/parser/ProjectResolver.java

```diff
@@ -18,6 +30,179 @@
 import com.microsoft.jdtls.ext.core.JdtlsExtActivator;
 
 public class ProjectResolver {
+
+    // Cache for project dependency information
+    private static final Map<String, CachedDependencyInfo> dependencyCache = new ConcurrentHashMap<>();
+    
+    // Flag to track if listeners are registered
+    private static volatile boolean listenersRegistered = false;
+    
+    // Lock for listener registration
+    private static final Object listenerLock = new Object();
+    
+    /**
+     * Cached dependency information with timestamp
+     */
+    private static class CachedDependencyInfo {
+        final List<DependencyInfo> dependencies;
+        final long timestamp;
+        final long classpathHash;
+        
+        CachedDependencyInfo(List<DependencyInfo> dependencies, long classpathHash) {
+            this.dependencies = new ArrayList<>(dependencies);
+            this.timestamp = System.currentTimeMillis();
+            this.classpathHash = classpathHash;
+        }
+        
+        boolean isValid() {
```

Comment:
what if in this 5 mins, user updated the original file for this cached dependency?

Reply:
Ps. Eclipse JDT will auto update the `.classpath` file once user update the dependency or something project settings.